sycl/doc/extensions/proposed/sycl_ext_oneapi_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,326 @@
+each other's local memory, and can be synchronized. This has various
+applications, convolutions, GEMMs and FFTs to name a few.
+
+This proposal introduces a SYCL equivalent of the same, to be called as
+`cluster_group`, and adds the methods to launch a kernel with a cluster range
+and accessing the various id's associated with the cluster
+launch in device code.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_CUDA_CLUSTER_GROUP` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not version-ed, so the
+ feature-test macro always has this value.
+|===
+
+
+=== Launching a kernel with a `cluster_group`
+
+Because of the special scheduling guarantees associated with a cluster launch,
+the backend must know which kernel would be using this feature. Thus, this 
+proposal introduces a new launch property called as `cluster_size` that will
+contain the cluster  in terms of number of workgroups.
+ 
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::cuda {
+/**
+* Dim Dimensionality of the launch
+* size sycl::range specifying the number of workgroups in the cluster
+  in each dimension.
+*/
+template <int Dim>
+struct cluster_size {
+  cluster_size(const sycl::range<Dim>& size) : 
+    size(size) {}
+  size_t operator[](int i) { return size[i]; }
+  sycl::range<Dim> size;
+};
+using cluster_size_key = cluster_size;
+} // namespace sycl::ext::oneapi::experimental::cuda
+----
+
+The property list can the be constructed as follows - 
+
+[source,c++]
+----
+properties  cluster_launch_property{cluster_size({1, 2, 1})};
+----
+
+Note that the total number of work-groups in the kernel must be a multiple of
+the cluster range in each dimension.
+
+The launch functions introduced in `sycl_ext_oneapi_enqueue_functions` can then
+be used to launch the kernel with the mentioned property.
+
+
+=== Accessing the Cluster Group From Device Code
+
+Building upon the group hierarchy in sycl, this proposal adds another level
+above `group`(for work-groups), to be called as `cluster-group`, which 
+represents a collection of work-groups and will be accessible via the `nd_item`
+class, via a method to be introduced called `ext_oneapi_cuda_get_cluster_group()`.
+
+
+[%header,cols="10,5"]
+|===
+
+|`cluster_group<Dimensions> nd_item::ext_oneapi_cuda_get_cluster_group()`
+|Returns the constituent `cluster_group` in the kernel, representing this
+`cluster_group`'s overall position in the `nd_range`
```

Comment:
This table renders slightly strangely
![image](https://github.com/intel/llvm/assets/2334151/ea825517-4896-42b5-beff-8841b2241976)

Might want this as the column headers in bold, also the monospace isn't quite right.
```
|Method
|Description

```

This should also be added to a couple other tables in this doc.